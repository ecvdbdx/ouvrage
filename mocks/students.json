[
  {
    "username": "epatrois",
    "first_name": "Manu",
    "last_name": "Patrois",
    "id": "2"
  },
  {
    "id": "5",
    "username": "mdewaele",
    "first_name": "Maxence",
    "last_name": "De Waele",
    "email": "maxencedewaele@gmail.com",
    "birthdate": "838921463",
    "avatar": "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"

  },
  {
    "id": "8",
    "username": "rodrigodelanoche",
    "first_name": "Vincent",
    "last_name": "Rodriguez",
    "email": "vincentrodrigu@hotmail.fr",
    "birthdate": "14/05/1413",
    "avatar": "http://etc.usf.edu/clipart/5500/5596/columbus_14_md.gif"
  },
  {
    "id": "6",
    "username": "erozis",
    "first_name": "Emilie",
    "last_name": "Rozis",
    "email": "emilie.40@hotmail.com"
  },
  {
    "first_name": "Hubert",
    "username": "hmoncenis",
    "id": "1",
    "last_name": "Montcenis"
  },
  {
    "id": "7",
    "username": "vivianeglz",
    "first_name": "Viviane",
    "last_name": "Gonzalez",
    "email": "viviane.gonzalez@outlook.fr",
    "birthdate": "02/06/1995",
    "avatar": "kikou.png"
  },
  {
    "id": "3",
    "username": "antoinesinety",
    "first_name": "Antoine",
    "last_name": "de Sinety",
    "email": "fake@yopmail.fr",
    "birthdate": "08/06/1995",
    "avatar": "Image.png"
  },
  {
    "id": "4",
    "username": "tgauzin",
    "first_name": "Thomas",
    "last_name": "Gauzin",
    "email": "thomas.gauzin@gmail.com"
  }
]
