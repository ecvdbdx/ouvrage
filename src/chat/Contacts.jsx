import React from 'react';

export default class Contacts extends React.Component {
  isContactActive (element) {
    if (element === this.props.currentContact) {
      return 'active';
    }
  }
  render () {
    return (
      <div id="list-conversations">
        <h2>Les contacts</h2>
        <ul>
          {
            this.props.students.map((student, index) => (
              <li key={index} onClick={() => this.props.openConversation(student)} className={this.isContactActive(student)}>
                <div className="contact-conv">
                  <img src={student.avatar} alt="Phot de profil de {student.username}"/>
                  <p>{student.first_name} {student.last_name} <span className="username">({student.username})</span></p>
                </div>
              </li>
            ))
          }
        </ul>
      </div>
    );
  }
}
