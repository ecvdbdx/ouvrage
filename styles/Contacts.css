.username {
  font-size: 13px;
  display: block;
}

#list-conversations h2 {
  margin: 50px 0;
  text-align: center;
  font-size: 25px;
}


#list-conversations li {
  display: block;
  width: 100%;
}

.contact-conv {
  display: flex;
  align-items: center;
  padding: 10px 20px;
}

#list-conversations img {
  border-radius: 50%;
  max-width: 50px;
  max-height: 50px;
  margin-right: 10px;
}

