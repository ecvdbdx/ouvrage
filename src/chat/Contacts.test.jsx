import React from 'react';
import renderer from 'react-test-renderer';
import students from './../../mocks/students.json';
import Contacts from './Contacts.jsx';
import { shallow } from 'enzyme';

const openConversationMock = jest.fn();

it('renders correctly', () => {
  const tree = renderer.create(<Contacts students={students} openConversation={openConversationMock}/>).toJSON();
  expect(tree).toMatchSnapshot();
});


it('open conversation on click', () => {
  const element = shallow(<Contacts students={students} openConversation={openConversationMock} />);
  element.find('li').at(0).simulate('click');
  expect(openConversationMock).toHaveBeenCalled();
});

