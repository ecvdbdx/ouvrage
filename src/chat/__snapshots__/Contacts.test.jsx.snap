// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="list-conversations"
>
  <h2>
    Les contacts
  </h2>
  <ul>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de epatrois"
          src="https://pbs.twimg.com/profile_images/781247417819725825/g8Rk4ljG_400x400.jpg"
        />
        <p>
          Manu
           
          Patrois
           
          <span
            className="username"
          >
            (
            epatrois
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de mdewaele"
          src="https://info.bestfriendspetcare.com/hubfs/BestFriendSpetcare-Sep2016/images/img_dog_right-col-top.png?t=1521138713469"
        />
        <p>
          Maxence
           
          De Waele
           
          <span
            className="username"
          >
            (
            mdewaele
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de rodrigodelanoche"
          src="https://s3.amazonaws.com/mixture-mixed/21975/17188/img/photos/portraits/dogs/golden-retriever.jpg"
        />
        <p>
          Vincent
           
          Rodriguez
           
          <span
            className="username"
          >
            (
            rodrigodelanoche
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de erozis"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSwMFoeG86Nz19Ve8Cf2FqpdiMd9Q8r6cYS1lgmPKZCxJfvb7kK"
        />
        <p>
          Emilie
           
          Rozis
           
          <span
            className="username"
          >
            (
            erozis
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de hmoncenis"
          src="https://barknbrewwi.com/wp-content/uploads/2017/05/bark-and-brew-dog-img.png"
        />
        <p>
          Hubert
           
          Moncenis
           
          <span
            className="username"
          >
            (
            hmoncenis
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de vivianeglz"
          src="https://img.freepik.com/free-photo/cute-golden-retriever-dog_1204-387.jpg?size=338&ext=jpg"
        />
        <p>
          Viviane
           
          Gonzalez
           
          <span
            className="username"
          >
            (
            vivianeglz
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de antoinesinety"
          src="https://cdn.petcarerx.com/LPPE/images/breedguidethumbs/Akita-dogs-photo.jpg"
        />
        <p>
          Antoine
           
          de Sinety
           
          <span
            className="username"
          >
            (
            antoinesinety
            )
          </span>
        </p>
      </div>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <div
        className="contact-conv"
      >
        <img
          alt="Photo de profil de tgauzin"
          src="https://cdn.petcarerx.com/LPPE/images/breedguidethumbs/Akita-dogs-photo.jpg"
        />
        <p>
          Thomas
           
          Gauzin
           
          <span
            className="username"
          >
            (
            tgauzin
            )
          </span>
        </p>
      </div>
    </li>
  </ul>
</div>
`;
